### ISSUE TITLE
Bi-directional pagination in ChannelPresenter

### ISSUE THREAD

**Issue by vadubov** [GetStream/stream-chat-swift#226]
In our app we create custom chat controller powered by ChannelPresenter. We need two features which seem to require custom initial pagination in ChannelPresenter:
1) restrict some users from seeing full chat history, meaning a user can see only messages with IDs or date greater than particular value;
2) initially load chat not at the latest messages, but somewhere in the middle of the history, by given message ID.
I figured it's impossible to implement using ChannelPresenter, though it can be done using Channel.query method, but we'd like to stick to ChannelPresenter. Do you plan to add initial pagination to ChannelPresenter in nearest future? Or maybe you could guide us to some workaround? Thanks

**Comment by b-onc** [comment]
Hello @vadubov ,

Thanks for explaining your usecases correctly.
Our function to start getting updates on channel (Presenter+Rx.swift:L49 - `prepareRequest(startPaginationWith:)`) accepts an initial pagination but our `ChannelPresenter` doesn't use this. The solution would be to add `messagesPagination:` and `repliesPagination:` parameter in `ChannelPresenter.init` and use these initial values to start request.

I'm keeping this issue as a feature request. If you'd like to implement and PR it, it'd be highly appreciated!

**Comment by b-onc** [comment]
Hello @vadubov 
I've pushed branch `226-initial-pagination-in-channelpresenter` with the changes. Can you test and let me know if that works for you?

**Comment by vadubov** [comment]
Hello @b-onc 
Thanks for your work, but it turns out that initial pagination is not enough for our purposes, it required implementing bi-directional pagination. I already implemented it in custom channel presenter, copypasting some code from `ChannelPresenter`, `Presenter`, `PresenterItem`, `ViewChanges`, and removing some unneeded for now features. I decided not to fork `StreamChatCore` since it would require more work and will contain breaking changes. This is my code, if you're interested in bidirectional pagination 
https://gist.github.com/vadubov/522719f94a5297c6f353b3a45b1de03c#file-customchannelpresenter-swift
